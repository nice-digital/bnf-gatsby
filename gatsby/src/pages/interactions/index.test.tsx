import { render, waitFor, screen } from "@testing-library/react";

import InteractionsIndexPage, { type InteractionsIndexPageProps } from "./";

const props: InteractionsIndexPageProps = {
	data: {
		allInteractants: {
			letters: [
				{
					letter: "a",
					links: [
						{
							title: "Abacavir",
							slug: "abacavir",
						},
						{
							title: "Abatacept",
							slug: "abatacept",
						},
					],
				},
				{
					letter: "b",
					links: [
						{
							title: "Bacillus Calmette-GuÃ©rin",
							slug: "bacillus-calmette-guerin-vaccine",
						},
						{
							title: "Bacitracin",
							slug: "bacitracin",
						},
					],
				},
			],
		},
	},
};

describe("InteractionsIndexPage", () => {
	it("should set page title", async () => {
		render(<InteractionsIndexPage {...props} />);
		await waitFor(() => {
			expect(document.title).toStartWith("Interactions A to Z");
		});
	});

	it("should set page meta description", async () => {
		render(<InteractionsIndexPage {...props} />);
		await waitFor(() => {
			expect(
				document
					// eslint-disable-next-line testing-library/no-node-access
					.querySelector("meta[name='description']")
					?.getAttribute("content")
			).toEqual(
				"Look up a drug to see which other drugs it interacts with and how serious those interactions are."
			);
		});
	});

	it("should match snapshot for page body", () => {
		render(<InteractionsIndexPage {...props} />);
		expect(screen.getByRole("main")).toMatchSnapshot();
	});
});
