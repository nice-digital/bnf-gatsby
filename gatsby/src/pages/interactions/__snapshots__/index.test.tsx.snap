// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InteractionsIndexPage should match snapshot for page body 1`] = `
<div>
  <nav
    aria-label="Breadcrumbs"
    role="navigation"
  >
    <ol
      class="breadcrumbs"
    >
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="https://www.nice.org.uk/"
        >
          NICE
        </a>
      </li>
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="/"
        >
          BNF
        </a>
      </li>
      <li
        class="breadcrumbs__crumb"
      >
        <span>
          Interactions A to Z
        </span>
      </li>
    </ol>
  </nav>
  <script
    type="application/ld+json"
  >
    {
  "@context": "http://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [
    {
      "@type": "ListItem",
      "position": 1,
      "item": {
        "@id": "https://www.nice.org.uk/",
        "name": "NICE"
      }
    },
    {
      "@type": "ListItem",
      "position": 2,
      "item": {
        "@id": "/",
        "name": "BNF"
      }
    },
    {
      "@type": "ListItem",
      "position": 3,
      "item": {
        "name": "Interactions A to Z"
      }
    }
  ]
}
  </script>
  <div
    class="page-header"
    id="content-start"
  >
    <h1
      class="page-header__heading "
    >
      Interactions
       A to Z
    </h1>
    <p
      class="page-header__lead"
    >
      Browse drugs by A to Z to check for interactions. Includes information on the severity of an interaction and the type of evidence to support it.
    </p>
  </div>
  <ol
    class="alphabet  "
    id="a-to-z"
  >
    <li
      class="alphabet__letter "
    >
      <a
        href="#a"
      >
        A
      </a>
    </li>
    <li
      class="alphabet__letter "
    >
      <a
        href="#b"
      >
        B
      </a>
    </li>
  </ol>
  <ol
    class="a-z-list "
  >
    <li
      class="a-z-list__item "
    >
      <h2
        class="a-z-list__item-heading"
        id="a"
      >
        A
      </h2>
      <ol
        aria-labelledby="a"
        class="column-list column-list--boxed"
        data-tracking="a-to-z-column-list"
      >
        <li>
          <a
            href="/interactions/abacavir/"
          >
            Abacavir
          </a>
        </li>
        <li>
          <a
            href="/interactions/abatacept/"
          >
            Abatacept
          </a>
        </li>
      </ol>
    </li>
    <li
      class="a-z-list__item "
    >
      <h2
        class="a-z-list__item-heading"
        id="b"
      >
        B
      </h2>
      <ol
        aria-labelledby="b"
        class="column-list column-list--boxed"
        data-tracking="a-to-z-column-list"
      >
        <li>
          <a
            href="/interactions/bacillus-calmette-guerin-vaccine/"
          >
            Bacillus Calmette-Guérin
          </a>
        </li>
        <li>
          <a
            href="/interactions/bacitracin/"
          >
            Bacitracin
          </a>
        </li>
      </ol>
    </li>
  </ol>
</div>
`;
